src/redis_bitmap.cc

```diff
@@ -79,6 +83,61 @@ rocksdb::Status Bitmap::GetBit(const Slice &user_key, uint32_t offset, bool *bit
   return rocksdb::Status::OK();
 }
 
+// Use this function after careful estimation, and reserve enough memory
+// according to the max size of the bitmap string to prevent OOM.
+rocksdb::Status Bitmap::GetString(const Slice &user_key, const uint32_t &max_btos_size, std::string *value) {
+  value->clear();
+  std::string ns_key, raw_value;
+  AppendNamespacePrefix(user_key, &ns_key);
+
+  BitmapMetadata metadata(false);
+  rocksdb::Status s = GetMetadata(ns_key, &metadata, &raw_value);
+  if (!s.ok()) return s;
+  if (metadata.size > max_btos_size) {
+    return rocksdb::Status::Aborted(kErrBitmapStringOutOfRange);
+  }
+  value->assign(metadata.size, 0);
+
+  std::string fragment, prefix_key;
+  fragment.reserve(kBitmapSegmentBytes * 2);
+  InternalKey(ns_key, "", metadata.version, storage_->IsSlotIdEncoded()).Encode(&prefix_key);
+
+  rocksdb::ReadOptions read_options;
+  LatestSnapShot ss(db_);
+  read_options.snapshot = ss.GetSnapShot();
+  read_options.fill_cache = false;
+  uint32_t frag_index, valid_size;
+
+  auto iter = db_->NewIterator(read_options);
+  for (iter->Seek(prefix_key);
+       iter->Valid() && iter->key().starts_with(prefix_key);
+       iter->Next()) {
+    InternalKey ikey(iter->key(), storage_->IsSlotIdEncoded());
+    frag_index = std::stoul(ikey.GetSubKey().ToString());
+    fragment = iter->value().ToString();
+    // To be compatible with data written before the commit d603b0e(#338)
+    // and avoid returning extra null char after expansion.
+    valid_size = std::min({fragment.size(),
+                          static_cast<size_t>(kBitmapSegmentBytes),
+                          static_cast<size_t>(metadata.size - frag_index)});
+
+    for (uint32_t i = 0; i < valid_size; i++) {
+        if (!fragment[i]) continue;
+        fragment[i] = ((fragment[i] & 0x80) >> 7)|\
```

Comment:
can help to explain why we need to swap the bit.

Reply:
Sorry for my late reply. Swapping bits is to get the same return value as Redis. This is like the endian conversion because Kvrocks setbit storage differs from Redis.

Reply:
Cool, how about adding a comment to explain the reason? it would be helpful when we can't recall the context.

Reply:
OK, I'll update some details tomorrow.